Migrations might fail or be incomplete due to reserved keywords

### Affected version(s)

4.13, 5.3

### Description

There is a curious issue where a migration might fail or be incomplete if you use something like this in your migration:

```php
// Get the table's columns
$columns = $schemaManager->listTableColumns($tableName);

// $columns will be an array<string, Column> where we expect 
// the array key to be the lowercased field name.
if (isset($columns[strtolower($fieldName)])) {
    // …
}
```

As it turns out the array key of `$columns` is not just a lowercased version of the field's name - but it might also be quoted according to the reserved keywords of your platform! (see [here](https://github.com/doctrine/dbal/blob/b05e48a745f722801f55408d0dbd8003b403dbbd/src/Schema/AbstractSchemaManager.php#L1415-L1416))

So for example, if you do

```php
$columns = $schemaManager->listTableColumns('tl_user');
```

then `$columns` will contain array keys like this:

```php
[
    'id' => …,
    'username' => …,
    '`admin`' => …,
]
```

under MySQL 8 - because `ADMIN` was a reserved keyword from MySQL 8.0.0 until 8.0.11 (it got [removed in MySQL 8.0.12](https://dev.mysql.com/doc/refman/8.0/en/keywords.html#keywords-8-0-detailed-A) - but `doctrine/dbal` never removed it).

Thus it is not save to just check simply via `isset($columns['foo'])` whether a column exists or not.

This causes our `BooleanFieldsMigration` to miss `tl_user.admin` under MySQL 8, resulting in an error like this

```
 * ALTER TABLE tl_user ADD doNotCollapse TINYINT(1) DEFAULT 0 NOT NULL, ADD frontendModules BLOB DEFAULT NULL, CHANGE showHelp showHelp TINYINT(1) DEFAULT 1 NOT NULL, CHANGE thumbnails thumbnails TINYINT(1) DEFAULT 1 NOT NULL, CHANGE useRTE useRTE TINYINT(1) DEFAULT 1 NOT NULL, CHANGE useCE useCE TINYINT(1) DEFAULT 1 NOT NULL, CHANGE pwChange pwChange TINYINT(1) DEFAULT 0 NOT NULL, CHANGE `admin` `admin` TINYINT(1) DEFAULT 0 NOT NULL, CHANGE disable disable TINYINT(1) DEFAULT 0 NOT NULL, CHANGE useTwoFactor useTwoFactor TINYINT(1) DEFAULT 0 NOT NULL......FAILED

 [ERROR] An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1366 Incorrect
         integer value: '' for column 'admin' at row 2
```

when updating from Contao 4.13 to 5.

--- a/core-bundle/src/Migration/Version500/BooleanFieldsMigration.php
+++ b/core-bundle/src/Migration/Version500/BooleanFieldsMigration.php
@@ -84,12 +84,11 @@ class BooleanFieldsMigration extends AbstractMigration
 
             foreach ($GLOBALS['TL_DCA'][$tableName]['fields'] ?? [] as $fieldName => $fieldConfig) {
                 $fieldName = strtolower($fieldName);
+                $field = $columns[$fieldName] ?? $columns["`$fieldName`"] ?? null;
 
-                if (!isset($columns[$fieldName]) || Types::BOOLEAN !== ($fieldConfig['sql']['type'] ?? null)) {
+                if (null === $field || Types::BOOLEAN !== ($fieldConfig['sql']['type'] ?? null)) {
                     continue;
                 }
-
-                $field = $columns[$fieldName];
 
                 if ($field->getType() instanceof StringType) {
                     $targets[] = [$tableName, $fieldName];

